src/main/webapp/app/shared/auth/RecaptchaBoundaryRoute.tsx

```diff
@@ -14,7 +21,42 @@ export interface IRecaptchaBoundaryRoute extends RouteProps {
 export class RecaptchaBoundaryRoute extends React.Component<
   IRecaptchaBoundaryRoute
 > {
+  public recaptcha = new ReCAPTCHA();
+  public recaptchaValidated: boolean;
+  public recaptchaError: any;
+
+  async loadData() {
+    try {
+      const token: string = await this.recaptcha.getToken();
+      setRecaptchaToken(token);
+      client
+        .validateRecaptchaUsingGET({})
+        .then(this.successToValidate, this.failedToValidate);
+    } catch (e) {
+      this.recaptchaError = e;
+    }
+  }
+
+  successToValidate() {
+    this.recaptchaValidated = true;
```

Comment:
This is a class and this variable is not a state. You might run into issue that the view is not updated when the variable is.